frb_codegen/src/commands.rs

```diff
@@ -18,11 +19,53 @@ fn call_shell(cmd: &str) -> Output {
     execute_command("sh", &["-c", cmd], None)
 }
 
-pub fn ensure_tools_available() -> Result {
-    let output = call_shell("dart pub global list");
-    let output = String::from_utf8_lossy(&output.stdout);
-    if !output.contains("ffigen") {
-        return Err(Error::MissingExe(String::from("ffigen")));
+pub fn ensure_tools_available(maybe_dart_root: &Option<String>) -> Result {
+    let dart_root = maybe_dart_root
+        .as_deref()
+        .unwrap_or(env!("CARGO_MANIFEST_DIR").into());
+    let toolchain = guess_toolchain(&dart_root).unwrap();
+    if toolchain == DartToolchain::Dart {
```

Comment:
uber-nit: `match toolchain`

Reply:
done in 374de6b36ce14583262ac744210d482c9553a2ae